src/main/java/com/microsoft/sqlserver/jdbc/SQLServerConnection.java

```diff
@@ -5829,6 +5831,39 @@ public void onEviction(Sha1HashKey key, PreparedStatementHandle handle) {
             }
         }
     }
+
+    boolean isAzureDW() throws SQLServerException, SQLException {
+        if (null == isAzureDW) {
+            try (Statement stmt = this.createStatement(); ResultSet rs = stmt.executeQuery("SELECT CAST(SERVERPROPERTY('EngineEdition') as INT)");)
+            {
+                // SERVERPROPERTY('EngineEdition') can be used to determine whether the db server is SQL Azure. 
+                // It should return 6 for SQL Azure DW. This is more reliable than @@version or serverproperty('edition').
+                // Reference:  http://msdn.microsoft.com/en-us/library/ee336261.aspx
+                // 
+                // SERVERPROPERTY('EngineEdition') means 
+                // Database Engine edition of the instance of SQL Server installed on the server.
+                // 1 = Personal or Desktop Engine (Not available for SQL Server.)
+                // 2 = Standard (This is returned for Standard and Workgroup.) 
+                // 3 = Enterprise (This is returned for Enterprise, Enterprise Evaluation, and Developer.)
+                // 4 = Express (This is returned for Express, Express with Advanced Services, and Windows Embedded SQL.)
+                // 5 = SQL Azure
+                // 6 = SQL Azure DW
+                // Base data type: int
+                final int ENGINE_EDITION_FOR_SQL_AZURE_DW = 6;
+                rs.next();
+                int engineEdition = rs.getInt(1);
```

Comment:
Change block to
`isAzureDW = rs.getInt(1) == ENGINE_EDITION_FOR_SQL_AZURE_DW`
or even more concise but possibly a little confusing
`return (rs.getInt(1) == ENGINE_EDITION_FOR_SQL_AZURE_DW)`

Reply:
i think right now is fine, i merely adopted the same part of code from our fx framework anyways.

Reply:
We should at least remove
`if (boolean condition) { `
`    bool = true`
`} else {`
`   bool = false`
`}`